src/main/java/org/assertj/core/api/recursive/comparison/RecursiveComparisonConfiguration.java

```diff
@@ -477,6 +505,10 @@ private boolean matchesAnIgnoredField(DualValue dualKey) {
     return matchesAnIgnoredField(dualKey.concatenatedPath);
   }
 
+  private boolean matchesAnIgnoredFieldType(DualValue dualKey) {
+    return ignoredFieldsForTypes.contains(dualKey.actual.getClass());
```

Comment:
`dualKey.actual` can be null, in that case it's a bit tricky, we could look at `dualKey.expected` but since we can compare objects of different types (ex: `Person` vs `PersonDto`) this might not be relevant.

Options when `dualKey.actual` is null:
* return false (don't ignore as we can't evaluate the type)
* evaluate `dualKey.expected` instead assuming `actual` and `expected` (not always true)
* ?

thoughts

Reply:
The use case for this method would be when someone needs to ignore a field with random values. 

As a user, I maybe not aware of the actual field is null, but I do want to ignore it whenever possible. 

In that sense, the second option is better, to check `dualKey.expected`'s type. And there are three scenarios: 

1. If the expected type matches, then obviously it should ignore (return true)
2. If the expected type does not match, it should not ignore (return false)
3. If both are null, then there is no need to ignore (return false)

User ignoring the types should be providing an expected value with the same type, so hopefully scenario 2 is unlikely to occur. And most of the time, the behaviour for handling null `dualKey.actual` should make sense. What do you think?

Reply:
> If the expected type matches, then obviously it should ignore (return true)
> If the expected type does not match, it should not ignore (return false)

we can't know that since if actual is null, you can't get its type

> If both are null, then there is no need to ignore as they are untyped and equal (return false)

I agree this is not an issue

Reply:
I think we should not ignore the field if actual is null otherwise we would ignore all null fields in actual.

Reply:
> I think we should not ignore the field if actual is null otherwise we would ignore all null fields in actual.

So evaluting`dualKey.expected` type will avoid ignoring all null fields, as in Scenario 1, you only ignore a null actual only when the expected has a matching type. 

Here is an example: 

I have a person with a pet: 
```
public class Person {
       private final String name;
       private final Pet pet;
}
```

I'd like to ignore the Pet, so I would do this: 

```
RecursiveComparisonConfiguration ignorePetConfig = new RecursiveComparisonConfiguration();
ignorePetConfig.ignoreFieldsForTypes(Pet.class);

assertThat(actual)
        .usingRecursiveComparison(ignorePetConfig)
        .isEqualTo(new Person("john", new Pet());
```

where `actual = new Person()` with null fields, so by checking the `dualKey.expected` type, it should only ignore the `pet` field, not the `name` field.  Do you see what I meant? 

It's basically what you suggest: 
`evaluate dualKey.expected instead assuming actual and expected (not always true)`

Reply:
what if you have a `PersonDto` with a `PetDto`?

Reply:
haven't used much DTO, but I'm taking the suggestion on returning false if the type cannot be evaluated, it's simple to reason about.